pkgs/development/libraries/volk/cpu-features-apple-silicon.patch

```diff
@@ -0,0 +1,276 @@
+--- a/cpu_features/CMakeLists.txt.orig
```

Comment:
Could you please send this patch upstream and fetch it using `fetchpatch`?

Reply:
I got this patch from the macports repo (see [here](https://github.com/macports/macports-ports/blob/master/science/volk/files/patch-cpu_features-add-support-for-ARM64.diff)).  I could fetch it from there, however, the patch did need some modification, I think the patch was applied at a different level of the tree and I had to prepend 'a/' and 'b/' to the file paths.

Reply:
The original patch was `-p0` and you need `-p1`. it was minimally sufficient to allow `cpu_features` to detect M1 / ARM64 and allow Volk to then build properly. It was a MacPorts-specific fix; there was a PR for upstream with a variant on this patch, but the `cpu_features` team took a different route that deprecated the need for my patch or the upstream PR.

Reply:
You can supply `extraPrefix = ""` to fetchpatch:
https://github.com/NixOS/nixpkgs/blob/7a632f5c1d698e75d10830a763fb2c9082c11b4d/pkgs/build-support/fetchpatch/default.nix#L36-L39
to convert it to be compatible with -p1 (don't forget to update the hash, or nix won't rebuild)

Reply:
Nice!  That worked great.